src/helm-frontend/src/types/Stat.ts

```diff
@@ -0,0 +1,27 @@
+export default interface Stat {
+  count: number;
+  max: number;
+  mean: number;
+  min: number;
+  name: {
+    name: string;
+    split?: string;
+    sub_split?: string;
+    perturbation?: {
+      computed_on: string;
+      fairness: boolean;
+      name: string;
+      name_file_path: string;
+      person_name_type: string;
+      preserve_gender: boolean;
+      prob: number;
+      robustness: boolean;
+      source_class: string;
+      target_class: string;
```

Comment:
I'm not sure if TypeScript supports this, but perturbation could contain extra fields of arbitrary type.

Also I think this definition can be combined with the `Perturbation.ts` earlier.

Reply:
I admittedly did a bit of best guessing with some of these types. Were you thinking about something like this?

```
export default interface Perturbation {
  [key: string]: string | number | boolean | undefined;
  name?: string;
  display_name?: string;
  description?: string;
  computed_on?: string;
  fairness?: boolean;
  name_file_path?: string;
  person_name_type?: string;
  preserve_gender?: boolean;
  prob?: number;
  robustness?: boolean;
  source_class?: string;
  target_class?: string;
}
```

And then using that type in `types/Stat`?

That would have the benefit of still providing some additional hinting within the IDE, while still allowing for additional values.

In a related note, I did notice that there are several typos where *Perturbation* is written as *Peturbation*. I will update that once we align on the best path forward here.

Reply:
Yes, I think `  [key: string]: string | number | boolean | undefined;` is a good idea. And it would be a good idea to resolve the typos as well.